lib/api/CDataFrameAnalysisRunner.cc

```diff
@@ -197,10 +199,18 @@ void CDataFrameAnalysisRunner::setToFinished() {
 
 CDataFrameAnalysisRunner::TStatePersister CDataFrameAnalysisRunner::statePersister() {
     return [this](std::function<void(core::CStatePersistInserter&)> persistFunction) -> void {
-        auto persistStream = m_Spec.persistStream();
-        if (persistStream != nullptr) {
-            core::CJsonStatePersistInserter inserter{*persistStream};
-            persistFunction(inserter);
+        auto persister = m_Spec.persister();
+        if (persister != nullptr) {
+            auto persistStream = persister->addStreamed(
+                ML_STATE_INDEX, m_Spec.jobId() + '_' + REGRESSION_TRAIN_STATE_TYPE);
```

Comment:
Let's just pass an empty string here instead of the index. The `CSingleStreamDataAdder` does not make use of the argument.

Reply:
I'd rather do this in a followup PR (which I can do at some point) that also changes other places where we reference the `.ml-state` index in the C++ code.  This method doesn't reference the `CSingleStreamDataAdder` class, so assuming it's the underlying implementation is breaching encapsulation and will trip up anyone who tries to use this code with a different type of data adder for test/debugging purposes.

Reply:
Sounds good.